resources/providers/gcplib/assets_inventory/provider.go

```diff
@@ -0,0 +1,171 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package assets_inventory
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/googleapis/gax-go/v2"
+	"google.golang.org/api/iterator"
+
+	asset "cloud.google.com/go/asset/apiv1"
+	"cloud.google.com/go/asset/apiv1/assetpb"
+
+	gcplib "github.com/elastic/cloudbeat/resources/providers/gcplib/auth"
+)
+
+type GcpAsset struct {
+	Type    string
+	SubType string
+
+	Asset *assetpb.Asset `json:"asset,omitempty"`
+}
+
+type Provider struct {
+	log    *logp.Logger
+	client Client
+	ctx    context.Context
+	config gcplib.GcpFactoryConfig
+}
+
+type Client interface {
+	ListAssets(ctx context.Context, req *assetpb.ListAssetsRequest, opts ...gax.CallOption) *asset.AssetIterator
+	Close() error
+}
+
+func NewAssetsInventoryProvider(ctx context.Context, log *logp.Logger, gcpConfig gcplib.GcpFactoryConfig) *Provider {
+	client, err := asset.NewClient(ctx, gcpConfig.ClientOpts...)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &Provider{
+		log:    log,
+		client: client,
+		ctx:    ctx,
+		config: gcpConfig,
+	}
+}
+
+func (p *Provider) ListAssets() ([]GcpAsset, error) {
+	var gcpAssets = make(map[string]*GcpAsset)
+
+	for typeName, Type := range gcpAssetGroups {
```

Comment:
I think that the provider methods should not be coupled to our use case.
I would modify this function to get asset types as an argument.